### PR TITLE
Add button

### PR DESCRIPTION
**Pull request by ArkadiyVoronov**
Add button to lib/senders/templates/new_device_login.html
Add some in lib/senders/partials/new_device_login.html

### CODE REVIEW

File: lib/senders/templates/new_device_login.html [hunk 1]
```diff
@@ -35,6 +35,29 @@ <h1 style="font-family: sans-serif; font-size: 21px; line-height: 29px; font-wei
...
+  <td align="center" valign="top">
+    <table border="0" cellpadding="0" cellspacing="0" height="100%" width="100%" id="email-button" style="-webkit-text-size-adjust: 100%; border-collapse: collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt; background-color: #0a84ff; border-radius: 4px; height: 50px; width: 310px !important;">
+      <tr style="page-break-before: always">
+        <td align="center" valign="middle" id="button-content" style="font-family: sans-serif; font-weight: normal; text-align: center; margin: 0; color: #ffffff; font-size: 20px; line-height: 100%;">
+          <!--[if mso]>
+          <v:roundrect xmlns:v="urn:schemas-microsoft-com:vml" xmlns:w="urn:schemas-microsoft-com:office:word" href="{{{link}}}" style="width:280px;height:40px;v-text-anchor:middle;" arcsize="10%" stroke="f" fillcolor="#0a84ff">
+            <w:anchorlock/>
+            <center>
+          <![endif]-->
+          <a href="{{{link}}}" id="button-link" style="font-family:sans-serif; color: #fff; display: block; padding: 15px; text-decoration: none; width: 280px; font-size: 18px; line-height: 26px;">{{t "Manage devices"}}</a>
```
Comment (vladikoff) [inline lib/senders/templates/new_device_login.html]: Please also add the button to the `.txt` version of the email here: https://github.com/mozilla/fxa-auth-server/blob/master/lib/senders/templates/new_device_login.txt

Existing Example: https://github.com/mozilla/fxa-auth-server/blob/f3261a613780cdec738069d262bc759c046041a3/lib/senders/templates/post_change_primary.txt#L5

File: lib/senders/partials/new_device_login.html [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
-<% block heading %>{{t "New sign-in to %(clientName)s" }}<% endblock %>
-<% block text_primary %>
+<% block content %>{{t "New sign-in to %(clientName)s" }}<% endblock %>
+<!--Header Area-->
+<tr style="page-break-before: always">
+  <td valign="top">
+    <h1 style="font-family: sans-serif; font-size: 21px; line-height: 29px; font-weight: normal; margin: 0 0 11px 0; text-align: center;">{{t "New sign-in to %(clientName)s" }}</h1>
+    <p class="primary" style="font-family: sans-serif; font-size: 14px; line-height: 21px; font-weight: normal; margin: 0 0 21px 0; text-align: center;">
+<p style="font-family:sans-serif; font-size: 13px; line-height: 20px; font-weight: normal; margin: 0 0 24px 0px; text-align: center; color: #4a4a4f;">
+    {{#if primaryEmail }}{{ primaryEmail }}<br/>{{/if}}
```
Comment (vladikoff) [inline lib/senders/partials/new_device_login.html]: these additions do not look correct, the template below already has  `<% include "./location/location.html" %>` included. You probably want this here: https://github.com/mozilla/fxa-auth-server/blob/f3261a613780cdec738069d262bc759c046041a3/lib/senders/partials/post_change_primary.html#L12-L32

File: lib/senders/templates/new_device_login.html [hunk 3]
```diff
@@ -35,6 +35,29 @@ <h1 style="font-family: sans-serif; font-size: 21px; line-height: 29px; font-wei
     {{#if timestamp }}{{ timestamp }}<br/>{{/if}}
 </p>
 
+<!--Button Area-->
+<tr height="50">
+  <td align="center" valign="top">
+    <table border="0" cellpadding="0" cellspacing="0" height="100%" width="100%" id="email-button" style="-webkit-text-size-adjust: 100%; border-collapse: collapse; mso-table-lspace: 0pt; mso-table-rspace: 0pt; background-color: #0a84ff; border-radius: 4px; height: 50px; width: 310px !important;">
+      <tr style="page-break-before: always">
+        <td align="center" valign="middle" id="button-content" style="font-family: sans-serif; font-weight: normal; text-align: center; margin: 0; color: #ffffff; font-size: 20px; line-height: 100%;">
```
Comment (vladikoff) [inline lib/senders/templates/new_device_login.html]: Please also add a test into this file: https://github.com/mozilla/fxa-auth-server/blob/bb2c67747b9ffe2dc398e4bfe7d22438c95cf3a8/test/local/senders/email.js#L206

you need a new `if` statement in there.

If email template is postChangePrimaryEmail or newDeviceLoginEmail.

Something like this (I haven't tested this code):
```
        if (type === 'postChangePrimaryEmail' || type === 'newDeviceLoginEmail' ) {
          it(
            'has manage account link in ' + type,
            function () {
              mailer.mailer.sendMail = function (emailConfig) {
                assert.ok(includes(emailConfig.html, 'Manage devices'))
                assert.ok(includes(emailConfig.text, 'Manage devices'))
              }
              mailer[type](message)
            }
          )
        }
```

File: lib/senders/templates/new_device_login.html (hunk 1 above)
Comment (vladikoff) [reply lib/senders/templates/new_device_login.html]: Also change the `link` variable in https://github.com/mozilla/fxa-auth-server/blob/a937c168f220996cb2094192663854f503f24c3b/lib/senders/email.js#L731 that should now be the same as:  https://github.com/mozilla/fxa-auth-server/blob/a937c168f220996cb2094192663854f503f24c3b/lib/senders/email.js#L836 (it should use `this.accountSettingsUrl`

**Comment by vladikoff** [comment]
@ArkadiyVoronov good start, I left some comments!

**Comment by ArkadiyVoronov** [comment]
Moved the project to the cloud. Made Grunt.

File: lib/senders/templates/new_device_login.html [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
   <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
-  <title>{{t "Firefox Accounts"}}</title>
+  <title>{{t "Manage account:"}} {{{ link }}}</title>
```
Comment (vladikoff) [inline lib/senders/templates/new_device_login.html]: @ArkadiyVoronov why are you changing the <title> tag ?

File: lib/senders/email.js [hunk 5]
```diff
@@ -728,7 +728,7 @@ module.exports = function (log, config) {
   Mailer.prototype.newDeviceLoginEmail = function (message) {
     log.trace({ op: 'mailer.newDeviceLoginEmail', email: message.email, uid: message.uid })
     var templateName = 'newDeviceLoginEmail'
-    var links = this._generateLinks(this.initiatePasswordChangeUrl, message.email, {}, templateName)
+    var links = this._generateLinks(this.accountSettingsUrl, message.email, {}, templateName)
```
Comment (vladikoff) [inline lib/senders/email.js]: this looks right now 👍

File: lib/senders/partials/new_device_login.html [hunk 6]
```diff
@@ -1,6 +1,33 @@
...
+ <!--Button Area-->
```
Comment (vladikoff) [inline lib/senders/partials/new_device_login.html]: Original file:

```html
<% extends "partials/base/base_text_only.html" %>

<% block heading %>{{t "New sign-in to %(clientName)s" }}<% endblock %>
<% block text_primary %>
<% include "./location/location.html" %>
<% endblock %>
```

The only change in this file you need is the `<tr>` row below, put that under `<% include "./location/location.html" %>`, you do not need the changes that you added above.

File: lib/senders/templates/new_device_login.html (hunk 3 above)
Comment (vladikoff) [reply lib/senders/templates/new_device_login.html]: @ArkadiyVoronov still need to the above to the email.js test file https://github.com/mozilla/fxa-auth-server/blob/bb2c67747b9ffe2dc398e4bfe7d22438c95cf3a8/test/local/senders/email.js#L206

File: lib/senders/templates/new_device_login.html (hunk 4 above)
Comment (ArkadiyVoronov) [reply lib/senders/templates/new_device_login.html]: I thought you had this line to fix it.

**Pull request closed without merging**